library/dash/src/main/java/com/google/android/exoplayer2/source/dash/manifest/DashManifestParser.java

```diff
@@ -1880,6 +1887,58 @@ protected static int parseMpegChannelConfiguration(XmlPullParser xpp) {
         : Format.NO_VALUE;
   }
 
+  /**
+   * Parses the number of channels from the value attribute of an AudioElementConfiguration with
+   * schemeIdUri as defined by Annex G (3.2) in ETSI TS 102 114 V1.6.1
+   *
+   * @param xpp The parser from which to read.
+   * @return The parsed number of channels, or {@link Format#NO_VALUE} if the channel count could
+   * not be parsed.
+   */
+  protected static int parseDtsChannelConfiguration(XmlPullParser xpp) {
+    @Nullable String value = xpp.getAttributeValue(null, "value");
+    if (value == null) {
+      return Format.NO_VALUE;
+    } else {
+      int channelCount = Integer.decode(value);
```

Comment:
Is there a particular reason to use `Integer.decode` rather than `Integer.parseInt`, given the spec defines the value as a decimal integer? I think this method can probably be written more simply as:

```
protected static int parseDtsChannelConfiguration(XmlPullParser xpp) {
  int channelCount = parseInt(xpp, "value", Format.NO_VALUE);
  return 0 < channelCount && channelCount < 33 ? channelCount : Format.NO_VALUE;
}
```

Reply:
I was just trying to be careful in case some non-standard packager put in the channel config "0x3f" instead of the channel count 6 . But it's ok, let's trust that the standards will be followed. Please replace decode with ParseInt.  Thanks.